service/resource/vaultcrt/create_test.go

```diff
@@ -128,3 +129,50 @@ func Test_Resource_VaultCrt_newCreateChange(t *testing.T) {
 		}
 	}
 }
+
+func Test_Resource_VaultCrt_ensureVaultRole_Organizations_intact(t *testing.T) {
+	testCases := []struct {
+		CustomObject          certificatetpr.CustomObject
+		ExpectedOrganizations []string
+	}{
+		// Test 0 ensures a non-nil current state results in the create state to be
+		// empty.
+		{
+			CustomObject: certificatetpr.CustomObject{
+				Spec: certificatetpr.Spec{
+					ClusterID:        "foobar",
+					ClusterComponent: "api",
+					Organizations:    []string{"system:masters"},
+				},
+			},
+			ExpectedOrganizations: []string{"api", "system:masters"},
+		},
```

Comment:
I would like to see multiple tests with different values, empty lists, lists with one and multiple items, also in different orders.